silkworm/infra/grpc/server/server.hpp

```diff
@@ -82,7 +82,7 @@ class Server {
         server_ = builder.BuildAndStart();
         SILK_TRACE << "Server " << this << " bound at selected port: " << selected_port;
         if (server_ == nullptr) {
-            SILK_ERROR << "Server " << this << " BuildAndStart failed [" << settings_.address_uri << "]";
+            SILK_ERROR << "Server " << this << " BuildAndStart failed. Cannot start gRPC server at [" << settings_.address_uri << "] Port is probably in use!";
```

Comment:
is there a way to get an error from the builder?

Reply:
I looked into exactly that. builder.BuildAndStart() just returns nullptr when an Error occurs.

Reply:
Try to implement [GlobalCallbacks::AddPort](https://github.com/grpc/grpc/blob/0b1e381d56da805045ce4c89d47ca4f13549c251/include/grpcpp/server.h#L86). Call `grpc::Server::SetGlobalCallbacks(...)` to register your implementation. `BuildAndStart()` calls it [here](https://github.com/grpc/grpc/blob/0b1e381d56da805045ce4c89d47ca4f13549c251/src/cpp/server/server_builder.cc#L434) and that should trigger a [callback](https://github.com/grpc/grpc/blob/master/src/cpp/server/server_cc.cc#L1131). If `port == 0` in the `AddPort` callback, we can assume that the port is busy.

Also please enhance the `std::runtime_error` thrown below with this extra information.